--- x264-snapshot-20121001-2245/common/cpu.c	2012-10-02 04:45:03.000000000 +0800
+++ x264-snapshot-20121001-2245.new/common/cpu.c	2012-10-31 12:00:35.075406933 +0800
@@ -31,6 +31,7 @@
 
 #if HAVE_POSIXTHREAD && SYS_LINUX
 #include <sched.h>
+#include <sys/syscall.h>
 #endif
 #if SYS_BEOS
 #include <kernel/OS.h>
@@ -403,10 +404,9 @@
     return sysconf( _SC_NPROCESSORS_ONLN );
 
 #elif SYS_LINUX
-    cpu_set_t p_aff;
-    memset( &p_aff, 0, sizeof(p_aff) );
-    if( sched_getaffinity( 0, sizeof(p_aff), &p_aff ) )
-        return 1;
+    int p_aff = 0;
+    if( syscall(__NR_sched_getaffinity, 0, sizeof(p_aff), &p_aff) )
+	return 1;
 #if HAVE_CPU_COUNT
     return CPU_COUNT(&p_aff);
 #else
