--- x264-snapshot-20121001-2245/configure	2012-10-02 04:45:03.000000000 +0800
+++ x264-snapshot-20121001-2245.new/configure	2012-10-31 12:05:19.303396233 +0800
@@ -765,7 +765,8 @@
             fi
             ;;
         *)
-            cc_check pthread.h -lpthread && thread="posix" && libpthread="-lpthread"
+	    # android has built-in pthread, do not need -pthread
+            cc_check pthread.h && thread="posix" && libpthread=""
             ;;
     esac
 fi
